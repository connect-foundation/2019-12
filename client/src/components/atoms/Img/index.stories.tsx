import React from 'react';
import { text } from '@storybook/addon-knobs';

import Img from '.';

export default {
  title: 'Atoms / Img',
};

export const index: React.FC = () => (
  <div style={{ padding: '1rem', backgroundColor: 'gray' }}>
    <div style={{ width: '100%' }}>
      <Img
        alt={text('alt', 'main banner img')}
        src={
          'https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1'
        }
      />
    </div>
    <div style={{ width: '100%' }}>
      <Img
        alt={text('alt', 'main banner img')}
        src={
          'https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-2'
        }
      />
    </div>
  </div>
);
