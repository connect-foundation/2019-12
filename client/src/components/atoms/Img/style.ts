import styled from 'styled-components';

export const Img = styled.img`
  width: inherit;
  height: inherit;
  object-fit: inherit;
`;
