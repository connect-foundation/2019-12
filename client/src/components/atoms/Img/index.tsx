import React, { useState } from 'react';

import * as S from './style';
import { useIntersect } from '../../../hooks';

export interface Props {
  /** alternation img */
  alt: string;
  /** image src */
  src: string;
  /** need Lazy Loading */
  lazy?: boolean;
}

function Img({ alt, src, lazy = true }: Props): React.ReactElement {
  const [imgSrc, setImage] = useState(
    lazy ? 'https://media.giphy.com/media/swhRkVYLJDrCE/source.gif' : src,
  );
  const [, setRef] = useIntersect(
    async (
      entry: IntersectionObserverEntry,
      observer: IntersectionObserver,
    ) => {
      setImage(src);
      observer.unobserve(entry.target);
    },
    {
      rootMargin: '0% 0% 33% 0%',
      threshold: 0.5,
    },
  );

  return <S.Img alt={alt} src={imgSrc} ref={lazy ? setRef : null} />;
}

export default Img;
