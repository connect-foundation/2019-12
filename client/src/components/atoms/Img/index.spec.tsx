import React from 'react';
import { mount } from 'enzyme';
import Img from '.';

describe('Atom / Img', () => {
  it('[SNAPSHOT] 렌더링', () => {
    // given
    const wrapper = mount(
      <Img
        alt="main banner img"
        src={
          'https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1'
        }
      />,
    );

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
