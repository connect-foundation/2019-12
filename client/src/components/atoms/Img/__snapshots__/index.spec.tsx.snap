// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom / Img [SNAPSHOT] 렌더링 1`] = `
.c0 {
  width: inherit;
  height: inherit;
  object-fit: inherit;
}

<Img
  alt="main banner img"
  src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
>
  <styled.img
    alt="main banner img"
    src="https://media.giphy.com/media/swhRkVYLJDrCE/source.gif"
  >
    <StyledComponent
      alt="main banner img"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  width: inherit;
  height: inherit;
  object-fit: inherit;
",
            ],
          },
          "displayName": "styled.img",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={[Function]}
      src="https://media.giphy.com/media/swhRkVYLJDrCE/source.gif"
    >
      <img
        alt="main banner img"
        className="c0"
        src="https://media.giphy.com/media/swhRkVYLJDrCE/source.gif"
      />
    </StyledComponent>
  </styled.img>
</Img>
`;
